### PR TITLE
Make error status messages user-friendly

### PR DESCRIPTION
**Pull request by deldesir**
This PR ensures the user gets comprehensible error messages. Should make debugging easier.

### CODE REVIEW

File: cps/tasks/metadata_extract.py [hunk 1]
```diff
@@ -71,14 +71,14 @@ def _fetch_requested_urls(self, conn):
             return {row[0]: {"duration": row[1], "is_playlist_video": self._is_playlist_video(row[0], conn)} for row in rows}
         except sqlite3.Error as db_error:
             log.error("An error occurred while trying to connect to the database: %s", db_error)
-            self.message = f"{self.media_url_link} failed: {db_error}"
+            self.message = f"{self.media_url_link} failed: An error occurred while trying to connect to the database."
```
Comment (holta) [inline cps/tasks/metadata_extract.py:74]: More complete is better!  If there's room??

```suggestion
            self.message = f"{self.media_url_link} failed: An error occurred ({db_error}) while trying to connect to the database."
```

**Comment by holta** [comment]
Fantastic news if "Tasks" view and log file explanations are becoming more meaningful 🔀 actionable.

(For dedicated, under resourced and rural educators / teachers / librarians especially!!)

**Comment by holta** [comment]
> Fantastic news if "Tasks" view and log file explanations are becoming more meaningful 🔀 actionable.
> 
> (For dedicated, under-resourced and rural educators / librarians / medics especially!!)

BONUS: If a 1st version of these UX and TDD improvements happens to be ready for the weekly build (weekly pre-releases) of IIAB Calibre-Web, that would be incredible:

Crystal clear + simple explanations are never as easy as they look — but are 💯% central to our global community challenge!🪜

**Comment by holta** [comment]
@deldesir can you resolve this merge conflict — and mention if this PR is fully ready / very safe to merge?

**Comment by deldesir** [comment]
Done

**Comment by EMG70** [comment]
Tested on a fresh VM 
SUDO IIAB-DIAGNOSTICS : https://dpaste.com/D9HL627NJ
The following were tested with varying results.
https://www.youtube.com/playlist?list=PLfiHW0cuXSvSiVuHqtO4IXMqWSrHou77B  ⚠️ 5/6 videos downloaded OK.Failed video's response "This is taking longer than expected"
This failed video https://www.youtube.com/watch?v=7qiwh4ybglo was tried individually and failed with "object cannot be interpreted as integer"
Screenshot from 2024-06-10 20-29-52

https://www.youtube.com/playlist?list=PLfiHW0cuXSvQ0s-oNi2syJNAPTN8gt4oi ❌ playlist of 29 videos failed
https://www.youtube.com/playlist?list=PLfiHW0cuXSvQSZHsA5zpo3JdrY5EJp1yT   ❌
https://www.youtube.com/watch?v=iaQV0tehds4 ✅
https://www.youtube.com/playlist?list=PLbeRJIjdJ7QEuVxTNfQsPBpUv4UcQdAzJ  ❌ short  playlist with 3 videos failed.

**Pull request merged**